pkg/sql/tests/autocommit_extended_protocol_test.go

```diff
@@ -156,18 +158,22 @@ func TestErrorDuringExtendedProtocolCommit(t *testing.T) {
 	var db *gosql.DB
 
 	var shouldErrorOnAutoCommit syncutil.AtomicBool
+	var traceID tracingpb.TraceID
 	params, _ := CreateTestServerParams()
 	params.Knobs.SQLExecutor = &sql.ExecutorTestingKnobs{
 		DisableAutoCommitDuringExec: true,
 		BeforeExecute: func(ctx context.Context, stmt string, descriptors *descs.Collection) {
 			if strings.Contains(stmt, "SELECT 'cat'") {
 				shouldErrorOnAutoCommit.Set(true)
+				traceID = tracing.SpanFromContext(ctx).TraceID()
 			}
 		},
 		BeforeAutoCommit: func(ctx context.Context, stmt string) error {
 			if shouldErrorOnAutoCommit.Get() {
 				shouldErrorOnAutoCommit.Set(false)
-				return errors.New("injected error")
+				if traceID == tracing.SpanFromContext(ctx).TraceID() {
```

Comment:
Can you add a comment explaining why the traceID needs to match?

Reply:
done